### PR TITLE
[Doc] Proof read faq.md

### PR DESCRIPTION
**Pull request by erizmr**
### Purpose

- Proof read the faq.md.

### Changes

- Polish the texts in the faq.md, delete content related to external libraries. 

### Review Instructions

- Please help review the new texts and point out any unclear/misleading descriptions. Thanks.

### CODE REVIEW

File: website/docs/lang/articles/faq.md [hunk 1]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
-## **Q:** Do we have something like `ti.pi`?
+## Does Taichi provide built-in constants such as `ti.pi`
 
-**A:** No, but you may use `math.pi` or `numpy.pi` instead. Taichi is
-able to bake in these constants during JIT, so your kernels incur no
-runtime cost.
+There is no built-in type for `pi` currently, `math.pi` is recommended to use in Taichi.
 
-## **Q:** How do I **force** an outermost loop to be serial, i.e. **not parallelized**?
+## How to **force** an outer-most loop to be serialized , i.e.,  **not parallelized**
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:15]:
```suggestion
## Outer-most loops in Taichi kernels are by default parallel. How can I **serialize** one of them?
```

File: website/docs/lang/articles/faq.md [hunk 2]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 
-**A:** Try this trick:
+A good practice is adding an additional `ghost` loop with only one iteration outside the loop you care about.
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:17]:
```suggestion
A good practice is adding an additional `ghost` loop with only one iteration outside the loop you want to serialize.
```

File: website/docs/lang/articles/faq.md [hunk 3]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 
 ```python {1}
-for _ in range(1):  # I'm the outer-most loop!
-    for i in range(100):  # This loop will not be parallelized
+for _ in range(1):  # The 'ghost' loop will be parallelized, but with only one thread i.e., serialized execution
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:20]:
```suggestion
for _ in range(1):  # This "ghost" loop will be "parallelized", but with only one thread. Therefore, the containing loop below is serialized.
```

File: website/docs/lang/articles/faq.md [hunk 4]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
+    for i in range(100):  # The loop you actually cares about will not be parallelized
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:21]:
```suggestion
    for i in range(100):  # The loop you want to serialize
```

File: website/docs/lang/articles/faq.md [hunk 5]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
         ...
 ```
 
-## **Q:** What's the most convenient way to load images or textures into Taichi fields?
+## What's the most convenient way to load images or textures into Taichi fields
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:25]:
```suggestion
## What is the most convenient way to load images or textures into Taichi fields?
```

File: website/docs/lang/articles/faq.md [hunk 6]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
 
 # Frequently Asked Questions
 
-## **Q:** Installing Taichi with `pip`, complains `package not found`.
+## My `pip` complains `package not found` when installing Taichi
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:7]:
```suggestion
## Why does my `pip` complain `package not found` when installing Taichi?
```

File: website/docs/lang/articles/faq.md [hunk 7]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 
-**A:** Simply use `field.from_numpy(ti.imread('filename.png'))`.
+One feasible solution is `field.from_numpy(ti.imread('filename.png'))`.
 
-## **Q:** Can Taichi co-operate with **other Python packages** like `matplotlib`?
+## Can Taichi interact with **other Python packages** such as `matplotlib`?
 
-**A:** Yes, as long as that _package_ provides an interface with
-`numpy`, see [Interacting with other Python packages](/docs/#interacting-with-other-python-packages).
+Yes, Taichi supports various popular Python packages, please check [Interacting with other Python packages](/docs/#interacting-with-other-python-packages).
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:31]:
```suggestion
Yes, Taichi supports various popular Python packages. Please check out [Interacting with other Python packages](/docs/#interacting-with-other-python-packages).
```

File: website/docs/lang/articles/faq.md [hunk 8]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 
-## **Q:** Shall we add some handy functions like `ti.smoothstep` or `ti.vec3`?
+## How to declare a field with **dynamic length**
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:33]:
```suggestion
## How do I declare a field with a **dynamic length**?
```

File: website/docs/lang/articles/faq.md [hunk 9]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
-**A:** You may export it with [Export PLY files](./misc/export_results#export-ply-files) so that you could view it in Houdini or Blender.
-
-:::tip
-Or make use the extension library [Taichi THREE](https://github.com/taichi-dev/taichi_glsl) to render images and update to GUI in real-time.
-:::
-
-## **Q:** How do I declare a field with **dynamic length**?
-
-**A:** What you want may be the `dynamic` SNode, a kind of sparse field, see [Working with dynamic SNodes](../api/snode.md#working-with-dynamic-snodes).
+The `dynamic` SNode supports variable-length fields. It serves as the role of `std::vector` in C++ or `list` in Python, please check [Working with dynamic SNodes](../api/snode.md#working-with-dynamic-snodes) for more details.
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:35]:
```suggestion
The `dynamic` SNode supports variable-length fields. It acts similarly to `std::vector` in C++ or `list` in Python. Please check out [Working with dynamic SNodes](../api/snode.md#working-with-dynamic-snodes) for more details.
```

File: website/docs/lang/articles/faq.md [hunk 10]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 
 :::tip
-Or simply allocate a dense field large enough, and another 0-D field
-`field_len[None]` for length record. But in fact, the `dynamic`
-SNode could be slower than the latter solution, due to the cost of
+An alternative solution is allocating a large enough `dense` field, with another 0-D field
+`field_len[None]` tracking its length. In practice, programs allocating memory using `dynamic`
+SNode may be less efficient than using `dense` SNode, due to the overhead of
 maintaining the sparsity information.
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:41]:
```suggestion
An alternative solution is to allocate a large enough `dense` field, with a corresponding 0-D field
`field_len[None]` tracking its length. In practice, programs allocating memory using `dynamic`
SNodes may be less efficient than using `dense` SNodes, due to dynamic data structure
maintainance overheads.
```

File: website/docs/lang/articles/faq.md [hunk 11]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 :::
 
-## **Q:** Can a user iterate over irregular topologies (e.g., graphs or tetrahedral meshes) instead of regular grids?
-
-**A:** These structures have to be represented using 1D arrays in Taichi. You can still iterate over them using `for i in x` or `for i in range(n)`.
+## How to  interact with irregular topologies (e.g., graphs or tetrahedral meshes)
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:44]:
```suggestion
## How do I interact with less structured data structures (such as graphs or tetrahedral meshes) in Taichi? 
```

File: website/docs/lang/articles/faq.md [hunk 12]
```diff
@@ -4,66 +4,43 @@ sidebar_position: 9999
...
 maintaining the sparsity information.
 :::
 
-## **Q:** Can a user iterate over irregular topologies (e.g., graphs or tetrahedral meshes) instead of regular grids?
-
-**A:** These structures have to be represented using 1D arrays in Taichi. You can still iterate over them using `for i in x` or `for i in range(n)`.
+## How to  interact with irregular topologies (e.g., graphs or tetrahedral meshes) 
 
-However, at compile time, there's little the Taichi compiler can do for you to optimize it. You can still tweak the data layout to get different runtime cache behaviors and performance numbers.
+These structures have to be stored in 1D fields in Taichi. You can traversal them using either `for element in x` or `for index in range(n)`.
```
Comment (yuanming-hu) [inline website/docs/lang/articles/faq.md:46]:
```suggestion
These structures have to be decomposed into 1D Taichi fields. For example, when representing a graph, you can allocate two fields, one for the vertices and the other for the edges. You can then traverse the elements using `for v in vertices` or `for v in range(n)`.
```

**Review by yuanming-hu — COMMENTED** [review]
Thank you! A few nits :-)

**Comment by erizmr** [comment]
Thanks for your useful suggestions! The doc looks more professional now.

**Pull request merged**